### ISSUE TITLE
Error 500 when trying to link GitHub Profile to existing Exercism profile

### ISSUE THREAD

**Issue by fapdash** [exercism/exercism#6557]
### Reproduction
1. Log in to Exercism
2. Navigate to https://exercism.org/settings
3. Click on the "Connect with GitHub" button
4. Give OAuth authorization on GitHub page

### Expected
After getting redirected to exercism.org the GitHub profile is linked to the currently logged in Exercism account.

### Actual
Error 500 error page shows up.
The profile is not linked to the account.

**Comment by iHiD** [comment]
Can you check that your GitHub account has a verified email please?

**Comment by fapdash** [comment]
@iHiD My GitHub account has several verified email addresses.
I also received an email from GitHub right after my first try:
> A third-party OAuth application (Exercism) with user:email scope was recently authorized to access your account.

So that part should be fine, GitHub also tells me that the OAuth access for exercism was used this week.

**Comment by iHiD** [comment]
OK, thanks. I can see you have a user with the handle `fap-` that's connected to that account, but that account was last updated in 2018, and that there is a different user with the handle `fap` that you've tried to connect today. So I think maybe the issue here is that you have two accounts. If you log out and log in via the GItHub auth action rather than email/password, do you find yourself logged into a different account?

**Comment by fapdash** [comment]
> If you log out and log in via the GItHub auth action rather than email/password, do you find yourself logged into a different account?

This leads to error 422.

Oh no, so I guess this still haunts me.
I had to change my GitHub username a few years ago because of a service that **only** supported log in via GitHub, but also didn't support all characters in their usernames that GitHub supports.

How do we untangle this? I don't see a possibility to unlink and relink on the web UI.

**Comment by fapdash** [comment]
I just checked https://exercism.org/contributing/contributors?track_slug=emacs-lisp&period=week and my account is showing up with my old GitHub name. Very curious..

My exercism account didn't receive any points from the merged PRs.
It also doesn't list the solutions that I published on the track.

![2022-09-28_22-15](https://user-images.githubusercontent.com/459631/192879936-27ee6d73-bbe4-4a1c-ab79-b595415908cc.png)

**Comment by fapdash** [comment]
@iHiD I figured it out!
The only way I found to make `User::LinkWithGithub` fail was an email conflict, see https://github.com/exercism/website/commit/15ff9c99b9be69c2dbb22b2b12514d14e95a2a80
I changed the email on my `fapdash` account. I then logged in in through GitHub via OAuth.
There actually was a second account and now I could properly log in without getting a 422 error, which probably was `devise` telling us about a unique email violation.
I deleted the account and connected the `fapdash` account to my GitHub and it worked. :)

I couldn't replicate the 422 error in the `User::AuthenticateFromOmniauthTest` so may be it was just a broken state that can't be reached through the current code.
I think it's not worth it to investigate further unless the error pops up again with another account.
Feel free to close if you concur.
